constellation_base/signals.py

```diff
@@ -0,0 +1,13 @@
+from django.dispatch import receiver
+from django.contrib.auth.signals import user_logged_in
+
+from .models import LocalGroupACLEntry
+
+
+@receiver(user_logged_in)
+def addSupplementalGroups(sender, user, request, **kwargs):
+    """Apply additional groups on login"""
+    supGroups = LocalGroupACLEntry.objects.filter(username=user.username)
+
+    for group in supGroups:
```

Comment:
You are proposing an implementation based on something like this:

`groupIter = subGroups.iterator()
while True:
    try:
        groupIter.next().group.user_set_add(user)
    except StopIteration:
      pass`